adr/0003-workflow-system.md

```diff
@@ -0,0 +1,58 @@
+# 3. Provide a workflow system
+
+Date: 2018-04-27
+
+## Status
+
+Proposed
+
+## Context
+
+In order to maintain portability and avoid dependency on any one service provider we wanted to explore solutions that did
+not depend on a specific platform.
+
+Although a Libero deployment should be workflow system agnostic, we want to provide a tried a tested solution as an option.
+
+## Decision
+
+There are many off the shelf solutions which offer vastly different levels of features and functionality.
+
+Each have their pros and cons, the key factors considered where:
+
+Usage:
+- Workflow and activity creation
+- Triggering workflows
+- Tracking workflow state
+
+Developer experience:
+- API complexity overhead
+- Minimal required system configuration
+- System deployment
+- System extension
+
+Integration:
+- How other services will communicate with it
+
+Libraries and frameworks that were considered:
+
+- [SpiffWorkflow](http://spiffworkflow.readthedocs.io/en/latest/)
+- [Toil](https://toil.readthedocs.io/en/3.15.0/)
+- [Luigi](http://luigi.readthedocs.io/en/stable/index.html)
+- [Conductor](https://netflix.github.io/conductor/)
+- [Airflow](https://airflow.incubator.apache.org/project.html)
+
+We also explored using a custom implementation consisting of the minimal components required to make up a lightweight workflow system written in pure Python.
+
+Our choice was to use [Airflow](https://airflow.incubator.apache.org/project.html) with a Libero wrapper to fulfil the required workflow integration points and manage the custom workflow and activities.
+
+## Consequences
+
+Only have to manage the code base for the wrapper and custom workflows and activities, not the overall system.
+
+Access to many pre made integrations with common services. 
+
+Must track progress of the [Airflow](https://airflow.incubator.apache.org/project.html) project and keep up to date.
+
+Handle the larger deployment requirements (when compared with other solutions).
```

Comment:
should we be worried about Airflow? it sounds large and unstable from your description.

Reply:
It was not meant to doubt the stability of the project, just to state that is a bit heavier on the deployment side due to dependencies than others we looked at (that said it is not much more from what you would expect), and that due to its [incubation status](https://airflow.apache.org/#apache-airflow-incubating-documentation) there are things that are going to change in the near future which may be of interest.

It seems to have a fairly large [user base](https://github.com/apache/incubator-airflow#who-uses-airflow) from the surface (though take from that what you will).

Reply:
It was a leading question. These ADRs needn't be so terse and formal we find ourselves having to read between the lines and asking what the author is _really_ trying to say.